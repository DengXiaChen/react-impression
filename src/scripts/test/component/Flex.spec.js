import React from 'react';
import { shallow, mount } from 'enzyme';
import { Flex } from '../../components/impression';

const setup = () => {
    const props = {
        align: 'middle',
        direction: 'column',
    };

    const wrapper = shallow(
        <Flex {...props}>
            <Flex.Item>Flex</Flex.Item>
        </Flex>
    );

    const wrapper1 = mount(
        <Flex {...props}>
            <Flex.Item>Flex</Flex.Item>
        </Flex>
    );

    return {
        props,
        wrapper,
        wrapper1,
    };
};

describe('Flex', () => {
    const { wrapper, wrapper1 } = setup();

    it('render', () => {
        expect(wrapper1.find('.flex.flex-vertical.flex-items-middle').exists()).toBe(true);
        expect(wrapper1.find('.flex-item').exists()).toBe(true);
    });

    it('ui', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
