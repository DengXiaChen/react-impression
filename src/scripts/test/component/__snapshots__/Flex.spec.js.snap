// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="flex flex-vertical flex-items-middle"
>
    <FlexItem
        flex={1}
    >
        Flex
    </FlexItem>
</div>,
  "nodes": Array [
    <div
      className="flex flex-vertical flex-items-middle"
>
      <FlexItem
            flex={1}
      >
            Flex
      </FlexItem>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Flex
        align="middle"
        direction="column"
>
        <FlexItem
                flex={1}
        >
                Flex
        </FlexItem>
</Flex>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Flex {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "align": "middle",
          "children": <FlexItem
            flex={1}
>
            Flex
</FlexItem>,
          "direction": "column",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="flex flex-vertical flex-items-middle"
>
          <FlexItem
                    flex={1}
          >
                    Flex
          </FlexItem>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="flex flex-vertical flex-items-middle"
>
          <FlexItem
                    flex={1}
          >
                    Flex
          </FlexItem>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Flex
    align="middle"
    direction="column"
>
    <FlexItem
        flex={1}
    >
        Flex
    </FlexItem>
</Flex>,
}
`;
